### PR TITLE
New Facebook and Web Messenger Cheat Sheet

### PR DESCRIPTION
**Pull request by saifakhan**
screen shot 2016-02-20 at 2 39 53 pm

This cheatsheet displays commands available to use on Facebook as well as the Web Messenger.

https://duck.co/ia/view/facebook_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@saifakhan Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@saifakhan apologies for the delay! Let's get this deployed to Beta for testing...

File: share/goodie/cheat_sheets/json/facebook.json [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"aliases": [
+		"Messenger",
+		"Web Messenger",
+		"Facebook Messenger",
+		"Facebook Web Messenger"
+	],
+	"sections": {
+		"Browser Keys": [{
+			"val": "Internet Explorer for PC",
+			"key": "Alt + #, then Enter"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/facebook.json]: @saifakhan please remove all the `+`. Individual keys should be wrapped in `[]`. Simultaneous key presses should just have a space ` ` between then. Consecutive ket presses should be separated with a comma `, `.

i.e. `"key": "Alt + #, then Enter"` should be `"key": "[Alt] [#], [Enter]"`

File: share/goodie/cheat_sheets/json/facebook.json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+			"key": "Ctrl + Opt + #"
+		}, {
+			"val": "Firefox for Mac",
+			"key": "Ctrl + Opt + #"
+		}, {
+			"val": "Chome for Mac",
+			"key": "Ctrl + Opt + #"
+		}],
+		"Access Keys": [{
+			"val": "0",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/facebook.json]: The are perfect as-is :+1: 
FYI single keys do ***not*** need to be wrapped in `[]`

File: share/goodie/cheat_sheets/json/facebook.json [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+			"key": "Search"
+		}, {
+			"val": "q",
+			"key": "Search chat contacts"
+		}, {
+			"val": "?",
+			"key": "Open a list of keyboard shortcuts while in the News Feed"
+		}],
+		"Web Messenger": [{
+				"val": "CTRL + g",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/facebook.json]: It looks like the indentation is off. Please make sure you indent the file wit 4-space soft tabs :+1:

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Facebook Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/facebook_cheat_sheet_mobile.png)

**Comment by DPAUTH** [comment]
Hi saifakhan great cheat sheet! But could you please add a source of where you got the cheat sheet. Plus in the Access keys section, news feed and web messengr I think you have misplaced the keys and the vals. Can't wait to see the cheat sheet go live!

File: share/goodie/cheat_sheets/json/facebook.json [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	],
+	"sections": {
+		"Browser Keys": [
+			{
+				"val": "Internet Explorer for PC",
+				"key": "[Alt] [#], [Enter]"
+			},
+			{
+				"val": "Firefox for PC",
+				"key": "Shift + Alt + #"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/facebook.json]: @saifakhan please remove all `+` -- any combined key presses should just have a space between them. Consecutive keys should have a `,` between them. All individual keys must be wrapped in `[]`. I.E:

`[Shift] [Alt] [#]`

File: share/goodie/cheat_sheets/json/facebook.json [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+			{
+				"val": "8",
+				"key": "About"
+			},
+			{
+				"val": "9",
+				"key": "Terms"
+			}
+		],
+		"News Feed": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/facebook.json]: The `val` and `key` here need to be switched

File: share/goodie/cheat_sheets/json/facebook.json [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+			{
+				"val": "q",
+				"key": "Search chat contacts"
+			},
+			{
+				"val": "?",
+				"key": "Open a list of keyboard shortcuts while in the News Feed"
+			}
+		],
+		"Web Messenger": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/facebook.json]: Same, the `val` and `key` for this section need to be switched

File: share/goodie/cheat_sheets/json/facebook.json [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+			{
+				"val": "j and k",
+				"key": "Scroll between News Feed stories"
+			},
+			{
+				"val": "enter/return",
+				"key": "See more of the selected story"
+			},
+			{
+				"val": "p",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/facebook.json]: BTW, these single-key commands **do not** need to be wrapped in `[]`

You just need to swap `key` and `val` here :+1:

**Comment by moollaza** [comment]
@saifakhan Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by saifakhan** [comment]
Hey @moollaza, I switched around the vals/keys as you indicated, sorry for not updating the brackets previously, you made a note on Access Keys and I thought it was for the remaining non-consecutive Browser Keys.

@DPAUTH, I've had a source in my JSON and the duckduckgo cheatsheet directory, is that not sufficient?

**Comment by saifakhan** [comment]
screen shot 2016-03-21 at 12 45 38 pm

Strange thing I'm noticing here, vals and keys are the same arrangement for both sections, yet Web Messenger looks incorrect. Any thoughts?

**Comment by alohaas** [comment]
@saifakhan thanks for your work on creating this new Facebook Messenger Cheat Sheet. It looks like you need some more review before we can merge it, so I'll see if I can't get @moollaza or another DDGer to help get this launched. You’ve been doing a really good job here and we'd like you to help make DuckDuckGo even better for yourself and others interested in Facebook and social media.

When you're ready to move on, please: 

- Create a new Instant Answer for searches you do related to Facebook or social media.  

OR

- Test live social media-related Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Creating-or-Improving-Instant-Answers).

Thanks again!

File: share/goodie/cheat_sheets/json/facebook.json [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+			},
+			{
+				"val": "Terms",
+				"key": "9"
+			}
+		],
+		"News Feed": [
+			{
+				"val": "Scroll between News Feed stories",
+				"key": "j and k"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/facebook.json]: this should be `[j] / [k]`

**Comment by moollaza** [comment]
@saifakhan Thanks! This LGTM :+1:

**Comment by daxtheduck** [comment]
@saifakhan this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Facebook Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/facebook_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Facebook Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/facebook_cheat_sheet_mobile.png)

**Pull request merged**